### PR TITLE
Bump: LH to v1.7.0-rc2

### PR DESCRIPTION
**Pull request by Yu-Jack**
**Solution:**
Try to bump it to v1.7.0-rc2 first.

**Major Bumps**
- longhorn v1.7.0-rc2, including deploy yaml.
- Bump golang to v1.22.5

**Related Bumping Dependencies Due to k8s.io/xxx**
- k8s.io/xxx v0.30.1
- kubevirt v1.3.0
- wrangler v3.0.0
- rancher/steve to v0.0.0-20240709130809-47871606146c

After bumping LH v1.7.0-rc1, although `go mod tidy` and `go mod vendor` are okay, but it encountered some issue while testing and building. So, I must bump above dependencies.

BTW, I'll squash commits while merging.

**Related Issue:**
https://github.com/harvester/harvester/issues/6172

### CODE REVIEW

**Comment by Yu-Jack** [comment]
Wait for build finishing.

**Review by FrankYang0529 — APPROVED** [review]
LGTM. VM basic operations work: migration / VMBackup / VMSnapshot / VMRestore.

**Comment by bk201** [comment]
@Yu-Jack Any reason to bump Kubevirt to v1.3.0?

**Comment by Yu-Jack** [comment]
> @Yu-Jack Any reason to bump Kubevirt to v1.3.0?

@bk201  Due to huge bumping of k8s.io/xxx series dependencies, lot of major dependencies also need bumping.

**Comment by WebberHuang1118** [comment]
> BTW, I'll squash commits while merging.

Since this is a huge hump, if possible please have a backup branch to keep the history before merging. We can refer to it afterward if needed. WDYT? @Yu-Jack

**Comment by Yu-Jack** [comment]
@WebberHuang1118 I mean I'll use `Squash and merge` button in Github GUI instead of squashing commits by myself. With this way, we still can check all commits after merged. You can check USB Passthrough PR on harvester/pcidevices as  reference.

**Comment by bk201** [comment]
@ Yu-Jack This PR is huge, can you squash the commits manually first before merging? It's OK to use the button for doc or something smaller.

**Comment by Vicente-Cheng** [comment]
Hi @Yu-Jack, could you try to squash manually to make the commits clearer for their target?

For example, you bump k8s 0.30, wrangler, kubevirt, and longhorn. Would it be good to have some commits for each component (like one for vendor bump/one for lint fix)?

That would be good for the reviewer or anyone wanting to know the history.

**Comment by Yu-Jack** [comment]
@Vicente-Cheng I'll try. If we bump only one dependency each commit, sometimes it failed to run `go mod tidy`. So, we need to bump multiple dependencies in one commits. But, I'll try to split commits and leave it failed status for better review experience.

**Comment by Vicente-Cheng** [comment]
> @Vicente-Cheng I'll try. If we bump only one dependency each commit, sometimes it failed to run `go mod tidy`. So, we need to bump multiple dependencies in one commits. But, I'll try to split commits and leave it failed status for better review experience.

Sure, Thanks for helping with that!
If there are too many dependencies, surely put them together.

**Comment by Yu-Jack** [comment]
@bk201 @Vicente-Cheng @WebberHuang1118 Please ignore those commits which contain `chore:` prefix. Those commits just produced some auto generated files. You can focus on other commits to see what I changed. Thanks.

**Comment by WebberHuang1118** [comment]
Thanks, @Yu-Jack for refactoring the commits, I checked basic operations. 
I will approve after double-checking the PR content.

The operations I checked under the storage network toward data integrity:
- Image
  - Uploading
  - Downloading
  - Exporting
- VM backup
- VM restore
  - Restoring to new VM
  - Restoring to existing VM
  - Restoring to new VM in different namespace
  - Restoring VM to another Harvester cluster

File: go.mod
Comment (WebberHuang1118) [inline go.mod]: Since we bump go.mod to kubevirt v1.3.0, don't we have to bump the chart to v1.3.0 like https://github.com/harvester/harvester/pull/6137?

Comment (Yu-Jack) [reply go.mod]: @WebberHuang1118 It's not prepared yet. So, we keep this as is.

**Review by WebberHuang1118 — APPROVED** [review]
LGTM, thanks for the LH bump.

**Comment by Vicente-Cheng** [comment]
Should we move to v1.7.0-rc2?

**Comment by Yu-Jack** [comment]
@Vicente-Cheng Done!

**Comment by Vicente-Cheng** [comment]
General LGTM, I will approve after some basic volume operations. Thanks for the effort!

**Review by Vicente-Cheng — APPROVED** [review]
LGTM, verified with all basic volume operations like create/delete/snapshot/backup and work well

**Comment by Yu-Jack** [comment]
@bk201 @FrankYang0529 @Vicente-Cheng @WebberHuang1118 I forgot to bump LH v1.7.0-rc2 charts, please see the last commit, thanks!

**Review by Vicente-Cheng — APPROVED** [review]
thanks!

**Review by FrankYang0529 — APPROVED** [review]
Thanks for update to v1.7.0-rc2.

**Comment by Vicente-Cheng** [comment]
Hi @bk201, @WebberHuang1118

We will merge it first for the coming sprint release.
Please check the latest update after you back.

**Comment by Yu-Jack** [comment]
@Mergifyio backport v1.4

**Comment by FrankYang0529** [comment]
Probably, we should squash to one commit next time.

**Comment by Yu-Jack** [comment]
cc @Vicente-Cheng

**Comment by Yu-Jack** [comment]
@bk201 I think we need to sync the convention about merging, should we use `squash and merge` or `rebase and merge` button in Github? It seems that some people use former, and some people use latter. In my opinion, I prefer to use `squash and merge`, it still keeps the full commits history in original PR and makes the commit more clear (because it's just a single commit in the end).

cc @FrankYang0529 @Vicente-Cheng

**Comment by bk201** [comment]
The team has been using rebase and merge to make the commit history look more clean. Squash and merge is intended for admin to use. But it's ok to use squash and merge in the doc repo, since it's quite common to have more commits after accepting requested changes.

For PR that has a lot of commits like this, it's better to do a manual squash for reviewer to review and let CI run again before merging. I think the suggestion from the reviewer is sane.

**Pull request merged**
